packages/teams-js/src/public/dialog.ts

```diff
@@ -103,12 +110,30 @@ export namespace dialog {
    */
   export function submit(result?: string | object, appIds?: string | string[]): void {
     ensureInitialized(FrameContexts.content, FrameContexts.sidePanel, FrameContexts.task, FrameContexts.meetingStage);
+
     if (!isSupported()) {
       throw errorNotSupportedOnPlatform;
     }
 
-    // Send tasks.completeTask instead of tasks.submitTask message for backward compatibility with Mobile clients
-    sendMessageToParent('tasks.completeTask', [result, appIds ? (Array.isArray(appIds) ? appIds : [appIds]) : []]);
+    app
+      .getContext()
+      .then((appContext) => {
+        // if actionInfo exists in appContext, the dialog was loaded from a 3rd party app,
+        // and if submit was called with a result, throw an error because the result cannot be
+        // passed along to the desired recipient
+        if (result && appContext.actionInfo) {
+          throw errorInvalidArguments;
```

Comment:
Throwing exceptions inside of a promise chain like this is not a good idea since there is no good way for developers to catch and handle them.

Reply:
We would need to change the return type of this function to `Promise<void>` to let the caller catch the error.

Reply:
When I first opened the PR it was returning `Promise<void>` and the caller was catching the error but I was asked to change it.  I assumed that rethrowing the error would allow it to be wrapped in a try/catch.  I will change it back to be more consistent with what is in the rest of the codebase.

Reply:
Thinking about it now, if I was using the api as is (not returning a promise), it wouldn't be immediately obvious to me to wrap this in a try catch, so I definitely agree